controllers/inventory-controller.js

```diff
@@ -37,6 +37,39 @@ const getAllInventories = async (_req, res) => {
   }
 };
 
+//Finding an inventory item with a specific id
+
+const getItemById = async (req, res) => {
+  try {
+    const itemFound = await knex("inventories")
+      .join("warehouses", "inventories.warehouse_id", "=", "warehouses.id")
+      .select(
+        "inventories.id",
+        "warehouses.warehouse_name",
+        "inventories.item_name",
+        "inventories.description",
+        "inventories.category",
+        "inventories.status",
+        "inventories.quantity"
+      )
+      .where("inventories.id", req.params.id);
+
+    if (itemFound.length === 0) {
+      return res.status(404).json({
+        message: `Inventory item with ID ${req.params.id} not found`,
```

Comment:
Great that we are handling on BE but its not being handled on FE. I tried reloading an inventory with an invalid Id. I did get 404 in the network tab but the UI was in permanent loading state. 
We should handle such API responses gracefully on FE. 
![image](https://github.com/ryn-ngn/instock-group-2-server/assets/8683116/77ef9e1b-3e16-4a68-8b02-c967ebf32e3d)